Import-LAPS.ps1

```diff
@@ -1,133 +1,132 @@
+$DownloadStartTime = Get-Date
 
-Invoke-WebRequest -Uri $url -OutFile $output
-Write-Output "Time taken: $((Get-Date).Subtract($start_time).Seconds) second(s)"
+Invoke-WebRequest -Uri $DownloadURL -OutFile $InstallationFilePath
+Write-Verbose -Message ('Time taken: {0} second(s)' -f ((Get-Date) - $DownloadStartTime).Seconds)
 
-# Install mgmt software on DC and write a verbose log of the install.
-msiexec /i "$env:homedrive\Shadow\LAPS\LAPSx64.msi" /passive /l*v "$env:homedrive\Shadow\LAPS\Install.log" ADDDEFAULT=ALL
+# Install management software on the DC and write a verbose log of the install.
+Start-Process -FilePath 'msiexec' -ArgumentList ('/i {0} /passive /l*v "{1}" ADDDEFAULT=ALL' -f $InstallationFilePath, (Join-Path -Path $WorkFolderPath -ChildPath $InstallLogFileName))
 
-# Import the new powershell cmdlets.
-# Update the AD schema to accomodate the new field for the password.
-
-Import-Module AdmPwd.PS
+# Import required PowerShell cmdlet.
+# Update AD schema to accomodate the new fields to store password data.
+Import-Module -Name AdmPwd.PS
 Update-AdmPwdADSchema
 
-# Get the default computer OU.
-# Reference https://support.microsoft.com/en-us/kb/324949
-$OUQuery = [adsisearcher]'(&(objectclass=domain))'
-$OUQuery.SearchScope = 'base'
-$OUQuery.FindOne().properties.wellknownobjects | ForEach-Object {
-	if ($_ -match '^B:32:AA312825768811D1ADED00C04FD8D5CD:(.*)$')
-	{
-		$Matches[1]
-	}
-}
-
-
 # Check if a computer OU was provided by the parameter and act accordingly.
 if (!$ComputerOU)
 {
-	Set-AdmPwdComputerSelfPermission -OrgUnit $Matches[1]
+	# Get the default computer OU.
+	# Reference https://support.microsoft.com/en-us/kb/324949
+	$OUQuery = [adsisearcher]'(&(objectclass=domain))'
+	$OUQuery.SearchScope = 'base'
+	$OUQuery.FindOne().properties.wellknownobjects | ForEach-Object {
+		if ($_ -match '^B:32:AA312825768811D1ADED00C04FD8D5CD:(.*)$')
+		{
+			$ComputerOU = $Matches[1]
+			Write-Verbose -Message $ComputerOU
+		}
+	}
 }
-else
+foreach ($OrgUnit in $ComputerOU)
 {
-	Set-AdmPwdComputerSelfPermission -OrgUnit $ComputerOU
+	Set-AdmPwdComputerSelfPermission -Identity $OrgUnit
 }
 
-# Configure who can read the attribute. By default only domain/enterprise admins can.
-if ($OrgUnitRead)
+# Configure who can read the password. By default only domain/enterprise admins can.
+if (!$OrgUnitRead)
 {
-	Set-AdmPwdReadPasswordPermission -OrgUnit $OrgUnitRead
+	$OrgUnitRead = $ComputerOU
 }
-if ($SecGroupRead)
-{
-	Set-AdmPwdReadPasswordPermission -AllowedPrincipals $SecGroupRead
+if ($SecGroupRead) {
+	foreach ($OrgUnit in $OrgUnitRead)
+	{
+		Set-AdmPwdReadPasswordPermission -Identity $OrgUnit -AllowedPrincipals $SecGroupRead
+	}
 }
 
 # Configure who can force a password change. By default only domain/enterprise admins can.
-if ($OrgUnitReset)
+if (!$OrgUnitReset)
 {
-	Set-AdmPwdResetPasswordPermission -OrgUnit $OrgUnitReset
+	$OrgUnitReset = $ComputerOU

```

Comment:
I'm unsure where you've moved the reset permissions for the OU to. The line you've removed set reset permissions to an OU (e.g.Ou that contains all service desk users) and it seems you're just assigning a variable here. Is this correct or have you moved it somewhere else?